cmake/thirdparty/SetupMPI.cmake

```diff
@@ -67,8 +69,24 @@ if (ENABLE_FIND_MPI)
     list(REMOVE_DUPLICATES _mpi_libraries)
 endif()
 
+# Allow users to override CMake's FindMPI
```

Comment:
Do we always want to override, as opposed to appending?

Reply:
I have run into cases where the variables are wrong and cannot be solved by appending.

Reply:
ok then.